File: bot.py
import feedparser
import requests
import os

TOKEN = os.getenv("8276947745:AAFwt9r7CnjGkzx_aNiYmiwf6B0QbDZ_9iA")
CHAT_ID = os.getenv("https://t.me/fftechsegredo")
RSS_URL = os.getenv("RSS_URL")

ARQUIVO = "ultimo.txt"

def enviar(msg):
    url = f"https://api.telegram.org/bot{TOKEN}/sendMessage"
    requests.post(url, data={"chat_id": CHAT_ID, "text": msg})

def ler_ultimo():
    try:
        with open(ARQUIVO, "r") as f:
            return f.read().strip()
    except:
        return ""

def salvar_ultimo(link):
    with open(ARQUIVO, "w") as f:
        f.write(link)

feed = feedparser.parse(RSS_URL)

if feed.entries:
    video = feed.entries[0]
    titulo = video.title
    link = video.link

    ultimo = ler_ultimo()

    if link != ultimo:
        mensagem = f"""🔥 NOVO VÍDEO!

{titulo}

📺 Assista agora:
{link}

🚀 Entra no canal pra mais dicas!"""
        
        enviar(mensagem)
        salvar_ultimo(link)

File: bot.js
const fetch = require("node-fetch");
const TOKEN = process.env.TOKEN;

console.log("Token existe?", TOKEN ? "SIM" : "NÃO"); // Linha de debug
console.log("Tamanho do token:", TOKEN?.length); // Tem que dar ~46

// resto do seu código...

// canais
const canais = [
  "@robinairdraktutoriais",
  "@afiliadosshopeebrl"
];

// produtos
const produtos = [
  "https://s.shopee.com.br/6ffHxillyX"
];

function montarMensagem(link) {
  return `🔥 OFERTA RELÂMPAGO 🔥

🛒 Produto na Shopee

🔗 ${link}

⚠️ Corre antes que acabe!`;
}

async function enviar(mensagem) {
  for (let canal of canais) {
    try {
      const res = await fetch(`https://api.telegram.org/bot${TOKEN}/sendMessage`, {
        method: "POST",
        headers: {
          "Content-Type": "application/json"
        },
        body: JSON.stringify({
          chat_id: canal,
          text: mensagem
        })
      });

      const data = await res.json();

      console.log("📤 Enviado para:", canal);
      console.log("📩 Resposta Telegram:", data);

    } catch (erro) {
      console.log("❌ Erro ao enviar para:", canal);
      console.log(erro);
    }
  }
}

async function rodar() {
  for (let link of produtos) {
    const msg = montarMensagem(link);
    await enviar(msg);
  }
}

rodar().catch(err => {
    console.error("ERRO GERAL:", err);
});